### PR TITLE
Dont use timer if closed

### PR DESCRIPTION
**Pull request by ohadbitt**

### CODE REVIEW

**Comment by yogilad** [comment]
Can you please add a description?
This is also so the change list for the version can be easily composed

**Review by yogilad — APPROVED** [review]

**Review by AsafMah — APPROVED** [review]

**Review by AsafMah — CHANGES_REQUESTED** [review]
Add to changelog.md

**Comment by ohadbitt** [comment]
> This is also so the change list for the version c

The description wouldve been a copy of the title

File: data/src/main/java/com/microsoft/azure/kusto/data/auth/PublicAppTokenProviderBase.java [hunk 1]
```diff
@@ -28,6 +28,11 @@ public abstract class PublicAppTokenProviderBase extends MsalTokenProviderBase {
         super(clusterUrl, authorityId, httpClient);
     }
 
+    @Override
+    protected synchronized String acquireAccessTokenImpl() throws DataServiceException, DataClientException {
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/auth/PublicAppTokenProviderBase.java:32]: Add a comment explaining this.

**Review by AsafMah — APPROVED** [review]
Approved - just add the comment, and make sure it actually does what it should (not making the thing pop up twice)

**Pull request merged**